fulp_modules/main_features/bloodsuckers/code/powers/feed.dm

```diff
@@ -194,133 +210,119 @@
 		else
 			to_chat(user, "<span class='warning'>Someone may have noticed...</span>")
 
-	else						 // /atom/proc/visible_message(message, self_message, blind_message, vision_distance, ignored_mobs)
+	else					// /atom/proc/visible_message(message, self_message, blind_message, vision_distance, ignored_mobs)
 		user.visible_message("<span class='warning'>[user] closes [user.p_their()] mouth around [target]'s neck!</span>", \
 						 "<span class='warning'>You sink your fangs into [target]'s neck.</span>")
-	// My mouth is full!
-	ADD_TRAIT(user, TRAIT_MUTE, BLOODSUCKER_TRAIT)
 
-	// Begin Feed Loop
-	var/warning_target_inhuman = FALSE
-	var/warning_target_dead = FALSE
-	var/warning_full = FALSE
-	var/warning_target_bloodvol = 99999
-	var/amount_taken = 0
-	var/blood_take_mult = amSilent ? 0.3 : 1 // Quantity to take per tick, based on Silent or not.
-	/// In a Frenzy? Take double the blood.
-	if(bloodsuckerdatum && bloodsuckerdatum.Frenzied)
-		blood_take_mult = 2
 	// Activate Effects
-	//target.add_trait(TRAIT_MUTE, BLOODSUCKER_TRAIT)  // <----- Make mute a power you buy?
+//	target.add_trait(TRAIT_MUTE, BLOODSUCKER_TRAIT)  // <----- Make mute a power you buy?
 
-	// FEEEEEEEEED!!! //
-	if(bloodsuckerdatum)
-		bloodsuckerdatum.poweron_feed = TRUE
-	while(target && active)
-		ADD_TRAIT(user, TRAIT_IMMOBILIZED, BLOODSUCKER_TRAIT) // user.canmove = 0 // Prevents spilling blood accidentally.
-		// Abort? A bloody mistake.
-		if(!do_mob(user, target, 2 SECONDS, extra_checks = CALLBACK(src, .proc/ContinueActive, user, target)))
-			// May have disabled Feed during do_mob
-			if(!active || !ContinueActive(user, target))
-				break
+	// FEEEEEEEEED!! //
+	bloodsuckerdatum?.poweron_feed = TRUE
+	ADD_TRAIT(user, TRAIT_MUTE, BLOODSUCKER_TRAIT) // My mouth is full!
+	ADD_TRAIT(user, TRAIT_IMMOBILIZED, BLOODSUCKER_TRAIT) // Prevents spilling blood accidentally.
```

Comment:
they had to stay still to finish that earlier do_after(); I think that if a bloodsucker tries to move at this point, it's because they do intend to break their feeding chain.

Reply:
They can cancel out at any time before the feed loop begins, after that they have to manually cancel, I find it weird to see someone drinking off of someone's neck as they're just wandering the halls.